proto/parallel/service.proto

```diff
@@ -6,8 +6,19 @@ syntax = "proto3";
 
 service Worker {
     rpc Submit(PickleData) returns (PickleData) {}
+    rpc Status(StatusRequest) returns (StatusResponse) {}
 }
 
 message PickleData {
     bytes binary = 1;
+}
+
+// Incase we want to send data in the future
+message StatusRequest{
+
+}
+
+message StatusResponse {
```

Comment:
@proteneer Let me know if anything else jumps out as important to verify.

Reply:
I think theses two are the big ones. There's more compiler specific things we may check later on:

- check the CUDA version that was used to compile the code (nvcc --version)
- check the CUDA_ARCH used in CMake

A more stringent/all-or-nothing approach might be to just check the hash of - I *think* they should generate identical byte code if the images are built identically.

``` bash
> md5sum custom_ops.cpython-37m-x86_64-linux-gnu.so
ae4aaa3d10e850372fdc0d51906000c7  custom_ops.cpython-37m-x86_64-linux-gnu.so
```

Reply:
I will investigate this further separate from this. We may run into issues with running this on heterogenous hardware as well, which perhaps this indicates is a bad idea?

Reply:
Agreed. Let's re-visit this later (if at all).

There are very small/subtle differences that may or may not matter between different GPUs/compilers, in fact one of those difference pertain to a technically incorrect but mostly inconsequential rounding issue I'm currently trying to address in #364. In general though, we probably can't guarantee bit-wise identical results across different GPUs (part of this probably has to do with our usage of cub).